frontend/src/components/codingPage/CodeDescArea.tsx

```diff
@@ -255,35 +231,11 @@ const CodeDescArea = (
 
         {/* Leaderboard */}
         <TabsContent value='leaderboard' data-testid="tabs-content-leaderboard" >
-            <div className='p-6' >
-                <Table data-testid="table" >
-                    <TableHeader>
-                        <TableRow>
-                            <TableHead className="w-[100px]"></TableHead>
-                            <TableHead>Name</TableHead>
-                            <TableHead>Points</TableHead>
-                            <TableHead className="text-right">Problem Solved</TableHead>
-                            <TableHead className="text-right">Runtime</TableHead>
-                        </TableRow>
-                    </TableHeader>
-
-                    <TableBody>
-                        {leaderboard.map((l, idx) => {
-                            return <TableRow key={`rank #${idx+1}`} >
-                                <TableCell className="font-medium">#{idx+1}</TableCell>
-                                <TableCell className='font-semibold text-primary'>{l.name}</TableCell>
-                                <TableCell className="text-right text-gray-500" >{l.points}</TableCell>
-                                <TableCell className="text-right text-gray-500" >{l.solved}</TableCell>
-                                <TableCell className="text-right text-gray-500" >{l.runtime}</TableCell>
-                        </TableRow>
-                        })}
-                    </TableBody>
-
-                    <TableFooter className='mt-3' >
-                        <TableRow><TableCell colSpan={5} className='text-gray-500' >{leaderboard.length} participant{leaderboard.length > 1 ? 's' : ''}</TableCell>
-                        </TableRow>
-                    </TableFooter>
-                </Table>
+            <div className='h-full p-6' >
+                <ScoreboardDataTable participants={leaderboard} />
+                <div className='mt-3 text-gray-500' >
+                    {leaderboard.length} participant{leaderboard.length > 1 ? 's' : ''}
```

Comment:
Nitpicky, but when leaderboard.length is 0, we still want the s (i.e. 0 participants)

```suggestion
                    {leaderboard.length} participant{leaderboard.length === 1 ? 's' : ''}
```